Javatrix.java

```diff
@@ -25,6 +25,50 @@
     int n;
 
     /**
+     * Construct a matrix from a 2D array.
+     *
+     * @param arr - 2D array to be copied into matrix
+     **/
+    public Javatrix(double[][] arr)
+    {
+        this.m = arr.length;
+        this.n = arr[0].length;
+        matrix = new double[m][n];
+        for (int i = 0; i < m; i++)
+        {
+            for (int j = 0; j < n; j++)
+            {
+                matrix[i][j] = arr[i][j];
+            }
+        }
+    }
+
+    /**
+     * Construct an x by y matrix from a 2D array.
+     * 
+     * @param arr - 2D array to be copied into matrix
+     * @param x - number of rows
+     * @param y - number of columns
+     */
+    public Javatrix(double[][] arr, int x, int y)
+    {
+        this.m = x;
+        this.n = y;
+        matrix = new double[m][n];
+        for (int i = 0; i < arr.length; i++)
+        {
+            for (int j = 0; j < arr[0].length; j++)
+            {
+                // If index not out of range for arr.
+                if (i < m && j < n)
```

Comment:
In that case, is it the expected behavior that it throws an ArrayIndexOutOfBoundsException when given x and y greater than the dimensions of the array?  If so should I re-write my test for that case to expect an exception, or should I remove it?